server/src/main/java/org/elasticsearch/index/mapper/FieldNamesFieldMapper.java

```diff
@@ -105,7 +103,9 @@ public FieldNamesFieldMapper build(BuilderContext context) {
                 String fieldName = entry.getKey();
                 Object fieldNode = entry.getValue();
                 if (fieldName.equals("enabled")) {
-                    builder.enabled(XContentMapValues.nodeBooleanValue(fieldNode, name + ".enabled"));
+                    deprecationLogger.deprecatedAndMaybeLog("field_names_enabled_parameter", ENABLED_DEPRECATION_MESSAGE);
+                    // just read the value and dump it on the floor
+                    XContentMapValues.nodeBooleanValue(fieldNode, name + ".enabled");
                     iterator.remove();
```

Comment:
I think we should still support this field (and log the deprecation) for indices created in a version where the option is available (7x) and throw an error if the version is on or after 8.0 ?